### PR TITLE
FINERACT-904 : Post Interest on Account Activation Date for Fixed Dep…

### PR DESCRIPTION
**Pull request by DheerajKumar19**
This PR implements
https://issues.apache.org/jira/browse/FINERACT-904

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-core/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsHelper.java [hunk 1]
```diff
@@ -161,6 +161,8 @@ private LocalDate determineInterestPostingPeriodEndDateFrom(final LocalDate peri
                 }
                 periodEndDate = periodEndDate.with(TemporalAdjusters.lastDayOfMonth());
             break;
+            case ACTIVATIONDATE:
```
Comment (adamsaghy) [inline fineract-core/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsHelper.java:164]: Can you please explain the math here?
Quote from the story:
_Assuming a customer's Fixed Deposit or Recurring Deposit account is activated on Jan 20, we post interest to the customer's account on Feb 20 and on the 20th of every subsequent month until the duration of the account lapses. So for a 6 month Fixed Deposit Account interest would be posted on Feb 20, Mar 20, Apr 20, May 20, June 20 and July 20._

So this is a kind of monthly interest posting...the difference is it is using the activation day + 1 month period, right? I am wondering whether an "ACTIVATION_DATE_MONTHLY" would be better name. Thoughts?
@bharathc27 FYI

Comment (DheerajKumar19) [reply fineract-core/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsHelper.java:164]: @adamsaghy Yes this is going to increment based on the activation day and increments monthly wrt date. As far as naming convention, please guide me here this is my first time around here at Fineract so please feel free to let me know what other changes I need to be doing here.

Comment (adamsaghy) [reply fineract-core/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsHelper.java:164]: My major concern is the story and the implementation is not matching... Please check the dates in the story and the dates in your test cases.

Comment (adamsaghy) [reply fineract-core/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsHelper.java:164]: @DheerajKumar19 Have you had the chance to check my concern regarding the implementation and the story requirement?

**Pull request closed without merging**